src/components/DatePickerRange.react.js

```diff
@@ -0,0 +1,306 @@
+            reopenPickerOnClearDates={ this.props.reopen_calendar_on_clear }
+            initialVisibleMonth={() => {
+              if (this.state.startDate != null) {
+                return this.state.startDate
+              } else {
+                return this.state.initialVisibleMonth
+              }
+            }}
+            numberOfMonths={ this.props.number_of_months_shown }
+            withPortal={ this.props.with_portal }
+            withFullScreenPortal={ this.props.with_full_screen_portal }
+            firstDayOfWeek={ this.props.first_day_of_week }
+            minimumNights={ this.props.minimum_nights }
+            enableOutsideDays={ this.props.show_outside_days }
+            monthFormat={ this.props.month_format }
+            displayFormat={ this.props.display_format }
+          />
+        );
+    }
+}
+
+DatePickerRange.propTypes = {
+    id: PropTypes.string,
+
+    /**
+     * Specifies the starting date for the component, best practice is to pass
+     * value via datetime object
+     */
+    start_date: PropTypes.string,
+
+    /**
+     * Specifies the ending date for the component, best practice is to pass
+     * value via datetime object
+     */
+    end_date: PropTypes.string,
+
+    /**
+     * Specifies the lowest selectable date for the component,
+     * best practice is to pass value via datetime object
+     */
+    min_date_range: PropTypes.string,
+
+    /**
+     * Specifies the highest selectable date for the component,
+     * best practice is to pass value via datetime object
+     */
+    max_date_range: PropTypes.string,
+
+    /**
+     * Specifies the date that is initially presented when the user
+     * opens the calendar, best practice is to pass value via datetime
+     * object
+     */
+    initial_visible_month: PropTypes.string,
+
+    /**
+     * Text that will be displayed in the first input
+     * box of the date picker when no date is selected - replaces 'Start Date'
+     */
+    start_date_placeholder_text: PropTypes.string,
+
+    /**
+     * Text that will be displayed in the second input
+     * box of the date picker when no date is selected - replaces 'End Date'
+     */
+    end_date_placeholder_text: PropTypes.string,
+
+    /**
+     * If True, the calendar will automatically open when cleared
+     */
+    reopen_calendar_on_clear: PropTypes.bool,
+
+    /**
+     * Number of calendar months that are shown when calendar is opened
+     */
+    number_of_months_shown: PropTypes.number,
+
+    /**
+     * If True, calendar will open in a screen overlay portal
+     */
+    with_portal: PropTypes.bool,
+
+    /**
+     * If True, calendar will open in a full screen overlay portal, will
+     * take precedent over 'withPortal' if both are set to true
+     */
+    with_full_screen_portal: PropTypes.bool,
+
+    /**
+     * Specifies what day is the first day of the week, values must be
+     * from [0, ..., 6]
```

Comment:
would it be helpful to say that `0` corresponds to Monday / Sunday? Or is that configurable? The day that starts the week is different around the world I believe